### DISCUSSION TITLE
how to pass variable to `dd` command

### DISCUSSION THREAD

**Discussion by gipsyh** [nushell/nushell#5392] [Q&A] [answered]
```
def main [name:string] {
    echo $name
    dd if=/dev/zero of=($name) bs=1024 count=1M
}
```

and then there is a file named `($name)` not $name

**Comment by kubouch** [comment, marked as answer]
Use string interpolation. Nushell doesn't expand strings automatically like other shells: https://www.nushell.sh/book/working_with_strings.html#double-quoted-strings

Something like `$'of=($name)'` should work.

**Reply by Abdillah** [reply]
I think the part "Nushell doesn't expand strings automatically like other shells" needs to be included in "Thinking in Nu". For new comer, it is important understanding this fact to replacing Bash for daily usage.

What's your take, @kubouch?

**Reply by kubouch** [reply]
Yeah, a short note in “Nushell isn't bash” would be useful, it is a common mistake.

**Reply by Abdillah** [reply]
Okay, with a set of example should be good then..

**Reply by goyalyashpal** [reply]
> it is a common mistake.

yep, was doing it again and again... the muscle memory would take some time to readjust
